### PR TITLE
[PURCHASE-1790] Add auction result filter instrumentation

### PR DESCRIPTION
**Pull request by sepans**
Fixes [PURCHASE-1790](https://artsyproduct.atlassian.net/browse/PURCHASE-1790)

Sample tracking events:

When size filter changes:
```JSON
{
  "context_page": "Artist Auction Results",
  "context_page_owner_id": "4d8b928b4eb68a1b2c0001f2",
  "context_page_owner_slug": "pablo-picasso",
  "context_page_owner_type": "Artist",
  "action_type": "Auction results filter params changed",
  "current": {
    "organizations": [],
    "sizes": [
      "MEDIUM"
    ],
    "openedItemIndex": null,
    "page": 1,
    "sort": "ESTIMATE_AND_DATE_DESC"
  },
  "changed": {
    "sizes": [
      "MEDIUM"
    ]
  }
}
```

When sort changes:
```JSON
{
  "context_page": "Artist Auction Results",
  "context_page_owner_id": "4d8b928b4eb68a1b2c0001f2",
  "context_page_owner_slug": "pablo-picasso",
  "context_page_owner_type": "Artist",
  "action_type": "Auction results filter params changed",
  "current": {
    "organizations": [],
    "sizes": [],
    "openedItemIndex": null,
    "page": 1,
    "sort": "ESTIMATE_AND_DATE_DESC"
  },
  "changed": {
    "sort": "ESTIMATE_AND_DATE_DESC"
  }
}
```

When a lot is expanded:
```JSON
{
  "context_page": "Artist Auction Results",
  "context_page_owner_id": "4d8b928b4eb68a1b2c0001f2",
  "context_page_owner_slug": "pablo-picasso",
  "context_page_owner_type": "Artist",
  "action_type": "Auction results filter params changed",
  "current": {
    "organizations": [],
    "sizes": [],
    "openedItemIndex": 2,
    "page": 1,
    "sort": "ESTIMATE_AND_DATE_DESC"
  },
  "changed": {
    "openedItemIndex": 2
  }
}
```

### CODE REVIEW

File: src/Apps/Artist/Routes/AuctionResults/AuctionResultsFilterContext.tsx [hunk 1]
```diff
@@ -29,6 +29,11 @@ export interface AuctionResultsFilterContextProps {
   setFilter: (name: keyof AuctionResultsFilters, value: any) => void
   unsetFilter: (name: keyof AuctionResultsFilters) => void
   onAuctionResultClick?: (index: number) => void
+  onFilterClick?: (
```
Comment (sepans) [inline src/Apps/Artist/Routes/AuctionResults/AuctionResultsFilterContext.tsx]: The `onFilterClick` ended up not being necessary for event tracking (see https://github.com/artsy/reaction/pull/3175) but I kept it here for future use though it doesn't have anything to do with instrumentation

File: src/Apps/Artist/Routes/AuctionResults/ArtistAuctionResults.tsx [hunk 2]
```diff
@@ -87,15 +90,15 @@ const AuctionResultsContainer: React.FC<AuctionResultsProps> = ({
...
-          //     AnalyticsSchema.ActionType.CommercialFilterParamsChanged,
-          //   current: filterContext.filters,
-          //   changed: {
-          //     [filterKey]: filterContext.filters[filterKey],
-          //   },
-          // })
+          tracking.trackEvent({
+            context_page: AnalyticsSchema.PageName.ArtistAuctionResults,
+            action_type:
+              AnalyticsSchema.ActionType.AuctionResultFilterParamChanged,
```
Comment (zephraph) [inline src/Apps/Artist/Routes/AuctionResults/ArtistAuctionResults.tsx]: Might need @louislecluse to verify this, but I'm not sure we should create a new page type for this tab. I _think_ it should just live under the artist page's context.

Comment (sepans) [reply src/Apps/Artist/Routes/AuctionResults/ArtistAuctionResults.tsx]: Forgot to link the slack discussion: https://artsy.slack.com/archives/C9YNS4X32/p1581631403052300?thread_ts=1581543838.020500&cid=C9YNS4X32

I might have misunderstood it though.

Comment (zephraph) [reply src/Apps/Artist/Routes/AuctionResults/ArtistAuctionResults.tsx]: Ahh, I see. All right then 👍

**Comment by louislecluse** [comment]
Thanks!
Can we change `action_type` to `event`? 
If you look at the filter change on the artwork grid, that is what is used for the event name.

Could you also tell me when this is in staging to QA?

Thanks!!

**Comment by sepans** [comment]
@louislecluse yes I can. But I followed the pattern that is used everywhere else in the code base (including artwork filter) so I suspect that `action_type` is converted to `event` somewhere down the tacking pipeline. @zephraph @damassi do you know if thats the case?

**Comment by zephraph** [comment]
@sepands, I believe that is the case, yes.

**Comment by artsyit** [comment]
:rocket: PR was released in `v25.12.0` :rocket:

**Pull request merged**